### PR TITLE
fix: add deprecation warnings to ultra legacy style, type and layout mixins + remove internal usage

### PR DESCRIPTION
**Pull request by mbylstra**
# Objective

Add deprecation warnings to mixins in:
- @kaizen/component-library/styles/type.scss
- @kaizen/component-library/styles/color.scss
- @kaizen/component-library/styles/layout.scss

Update all internal imports of these files to the respective (and identical) files in `@kaizen/deprecated-component-library-helpers/styles`

# Motivation and Context

Kaizen currently contains duplicate files for `type.scss`, `style.scss` and `layout.scss`. One set is under `@kaizen/component-library/styles`, and the other is under `@kaizen/deprecated-component-library-helpers/styles`.

Confusingly, the ones in `@kaizen/component-library `are _more_ discouraged than the ones in `@kaizen/deprecated-component-library-helpers` and they certainly should not both exist. Code duplication (I mean where code is objectively and _exactly_ duplicated, not the subjective kind of code duplication)  is one of the worst forms of technical debt IMO.

This PR should also speed up compilation as there were many files that were importing both files, and needlessly compiling the second one.

# Details

After adding the deprecatation warnings it was a apparent that kaizen itself contained hundreds of (indirect) references to these legacy mixins, causing thousands of scss compilation warnings which would slow down compilation (and apparently. They were helpful in removing the references, but unfortunately scss error messages are very bad due to the lack of a proper import system so they don't tell exactly which file is responsible (as we can't use scss's more recent module system, as we would need to use dart scss, which we abandoned hope of being able to use). It was a lot of searching, trial and error trying to figure out which files had the imports.

Example warning:
```
WARNING: @kaizen/component-library/styles/type is completely deprecated. Instead, please use one of the following in order from ideal to 'if you really must': 1: use the Heading and Paragraph components, 2: use the `kz-typography-` mixins
 in @kaizen/deprecated-component-library-helpers/styles/type, 3: use the `ca-type-` mixins in @kaizen/deprecated-component-library-helpers/styles/type which are identical to these deprecated mixins
         on line 41 of packages/component-library/styles/type.scss, in mixin `deprecate-legacy-scss-typography-mixins`
         from line 87 of packages/component-library/styles/type.scss, in mixin `ca-inherit-baseline`
         from line 10 of stdin
```

# Out of Scope

- scss variable deprecation warnings. There is no way to do that in scss unfortunately. More info here: https://github.com/cultureamp/kaizen-design-system/issues/701#issuecomment-782972132

# Screenshots (if appropriate)

# Checklist

- [x] If this contains visual changes, has it been reviewed by a designer? (no visual changes expected)
- [x] If this introduces either a new component or a breaking change to an existing component, has it been reviewed by an advocate? (Ask in #prod_design_systems)
- [x] I have considered likely risks of these changes and got someone else to QA as appropriate - checking in Chromatic seems adequate - green build here: https://www.chromatic.com/build?appId=5e12c297b9d9020020789ca3&number=256
- [x] I have or will communicate these changes to the front end practice

# Additional Considerations
- Does there need to be right-to-left (RTL) options for localization and internationalization?
- Has the test suite been updated?
- Have you done cross-browser testing for our [supported browsers](https://academy.cultureamp.com/hc/en-us/articles/204539569-Supported-browsers-for-Participants)?
- Have you updated any relevant documentation or left useful comments in the code?
- Have you reviewed Culture Amp's Web Accessibility guide [Current Compliance, Going Forward, Statement and Answers for Customers](https://cultureamp.atlassian.net/wiki/spaces/Prod/pages/428572998/Web+Accessibility)?
- If this contains substantial visual changes, especially far reaching ones, have you unlocked the Chromatic step in the pipeline?
- Have Storybook stories been updated?

**If you're new to Kaizen, please ask #prod_design_systems to set up an onboarding session to get you up to speed.** If you have an urgent PR to merge before that happens, it is safest to ask in #prod_design_systems to have a design system advocate review the PR to catch any issues.

### CODE REVIEW

File: draft-packages/form/KaizenDraft/Form/TextField/styles.scss [hunk 1]
```diff
@@ -1,7 +1,8 @@
 @import "~@kaizen/design-tokens/sass/color-vars";
 @import "~@kaizen/component-library/styles/animation";
-@import "~@kaizen/component-library/styles/layout";
-@import "~@kaizen/component-library/styles/type";
+@import "~@kaizen/deprecated-component-library-helpers/styles/layout";
+@import "~@kaizen/deprecated-component-library-helpers/styles/type";
+@import "~@kaizen/deprecated-component-library-helpers/styles/color";
```
Comment (jasononeil) [inline draft-packages/form/KaizenDraft/Form/TextField/styles.scss:5]: Good pickup that color import was missing. SASS 😡

File: packages/component-library/styles/color.scss [hunk 2]
```diff
@@ -2,15 +2,20 @@
 $white: white;
 $black: black;
 
+$deprecation-warning: "@kaizen/component-library/styles/color is completely deprecated. Please use tokens from @kaizen/design-tokens instead. Or if you really must, use the mixins in @kaizen/deprecated-component-library-helpers/styles/color instead.";
+
 @function add-tint($color, $percentage) {
+  @warn $deprecation-warning;
```
Comment (jasononeil) [inline packages/component-library/styles/color.scss:8]: Let's make sure this doesn't cause murmur to collapse

**Review by jasononeil — COMMENTED** [review]
Let's check a yarn link or canary release doesn't cause murmur to explode. Other repos should be fine.

Also
- [ ] Make a commit that deletes the old files
- [ ] Fix any resulting errors
- [ ] Revert that commit (we'll save the breaking change for later)

**Comment by mbylstra** [comment]
I though about yarn linking or using , but then I realised I'd need to do that for every single draft package (and component-library package), which would be quite a lot of work.

I did a search for imports in murmur, and only counted 19. I don't think it will slow down scss compilation much. The big problems were when Kaizen itslef was generating warnings, and some of those files were imported thousands of times as indirecte dependencies. The 19 or so imports were direct imports that would cause warnings, and say each file had 2 or so warnings, there should be more than a 100 warnings (which are also very easy to fix - I'm happy to help with updating Murmur before we release the codemod PR if it's causing annoyance).

**Review by jasononeil — APPROVED** [review]

**Pull request merged**